Source/carthage/CopyFrameworks.swift

```diff
@@ -165,7 +165,8 @@ private func inputFiles() -> SignalProducer<String, CarthageError> {
 				return SignalProducer(result: getEnvironmentVariable("SCRIPT_INPUT_FILE_\(index)"))
 			}
 
-			return SignalProducer<SignalProducer<String, CarthageError>, CarthageError>(variables)
-				.flatten(.concat)
+			return SignalProducer.zip(variables)
+				.map { Set($0) }
+				.flatMap(.concat) { SignalProducer<String, CarthageError>($0) }
```

Comment:
Would you mind cleaning this up a bit and using `uniqueValues`? Something like this:

```swift
return SignalProducer(result: count)
    .flatMap(.merge) { return .init(values: 0..<$0) }
    .map { getEnvironmentVariable("SCRIPT_INPUT_FILE_\($0)" }
    .uniqueValues()
```